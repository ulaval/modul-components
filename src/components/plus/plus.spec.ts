import { mount, Wrapper } from '@vue/test-utils';
import Vue, { VueConstructor } from 'vue';

import { renderComponent } from '../../../tests/helpers/render';
import PlusPlugin, { MPlus } from './plus';

describe('MPlus', () => {
    let localVue: VueConstructor<Vue>;
    beforeEach(() => {
        Vue.use(PlusPlugin);
    });

    it('should render correctly', () => {
        const component: Wrapper<MPlus> = mount(MPlus, {
            localVue: localVue
        });

        return expect(renderComponent(component.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly with open prop', () => {
        const component: Wrapper<MPlus> = mount(MPlus, {
            localVue: localVue,
            propsData: {
                open: true
            }
        });
        expect(component.classes()).toContain('m--is-open');
        return expect(renderComponent(component.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly with large prop', () => {
        const component: Wrapper<MPlus> = mount(MPlus, {
            localVue: localVue,
            propsData: {
                large: true
            }
        });
        expect(component.classes()).toContain('m--is-large');
        return expect(renderComponent(component.vm)).resolves.toMatchSnapshot();
    });
    it('should render correctly with border prop', () => {
        const component: Wrapper<MPlus> = mount(MPlus, {
            localVue: localVue,
            propsData: {
                border: true
            }
        });
        expect(component.classes()).toContain('m--has-border');
        return expect(renderComponent(component.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly with disabled prop', () => {
        const component: Wrapper<MPlus> = mount(MPlus, {
            localVue: localVue,
            propsData: {
                disabled: true
            }
        });
        expect(component.classes()).toContain('m--is-disabled');
        return expect(renderComponent(component.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly with skin prop light', () => {
        const component: Wrapper<MPlus> = mount(MPlus, {
            localVue: localVue,
            propsData: {
                skin: 'light'
            }
        });

        expect(component.classes()).toContain('m--is-skin-light');
        return expect(renderComponent(component.vm)).resolves.toMatchSnapshot();
    });

    it('should emit click event when m-plus is clicked on', () => {
        const component: Wrapper<MPlus> = mount(MPlus, {
            localVue: localVue
        });

        component.trigger('click');

        expect(component.emitted('click'));
    });
});
