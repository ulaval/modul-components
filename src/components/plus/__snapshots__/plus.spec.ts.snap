// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MPlus should render correctly 1`] = `<span class="m-plus m--is-skin-default"></span>`;

exports[`MPlus should render correctly with border prop 1`] = `<span class="m-plus m--is-skin-default m--has-border"></span>`;

exports[`MPlus should render correctly with disabled prop 1`] = `<span class="m-plus m--is-skin-default m--is-disabled"></span>`;

exports[`MPlus should render correctly with large prop 1`] = `<span class="m-plus m--is-skin-default m--is-large"></span>`;

exports[`MPlus should render correctly with open prop 1`] = `<span class="m-plus m--is-skin-default m--is-open"></span>`;

exports[`MPlus should render correctly with skin prop light 1`] = `<span class="m-plus m--is-skin-light"></span>`;
